Source/WebCore/PAL/pal/text/TextCodecUTF16.cpp

```diff
@@ -66,14 +66,11 @@ void TextCodecUTF16::registerCodecs(TextCodecRegistrar registrar)
     });
 }
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-
 // https://encoding.spec.whatwg.org/#shared-utf-16-decoder
 String TextCodecUTF16::decode(std::span<const uint8_t> bytes, bool flush, bool, bool& sawError)
 {
-    const auto* p = bytes.data();
-    const auto* const end = p + bytes.size();
-    const auto* const endMinusOneOrNull = end ? end - 1 : nullptr;
+    size_t index = 0;
+    const size_t lengthMinusOne = bytes.size() - 1;
```

Comment:
```suggestion
    size_t lengthMinusOne = bytes.size() - 1;
```
So many local variables _could_ be marked `const` but we generally don't do it, for brevity's sake. Although I see that the old code did.